File: unit-3/sprint-1/day-6_View Pager/we/Send Data from one tab to another/app/src/main/java/com/example/complex/MainActivity.java
package com.example.complex;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentStatePagerAdapter;
import androidx.viewpager.widget.ViewPager;
import android.os.Bundle;
import com.google.android.material.tabs.TabLayout;
public class MainActivity extends AppCompatActivity {
    private ViewPager viewPager;
    private TabLayout tabLayout;
    private String name;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        viewPager = findViewById(R.id.viewPager);
        tabLayout = findViewById(R.id.tabLayout);
        setViewPagerAdapter();
    }

    private void setViewPagerAdapter() {
        ViewPagerAdapter viewPagerAdapter = new ViewPagerAdapter(getSupportFragmentManager(),
                FragmentStatePagerAdapter.BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT);
        viewPager.setAdapter(viewPagerAdapter);
        tabLayout.setupWithViewPager(viewPager);
    }

    public void setData(String name) {
        this.name = name;
    }

    public void sendDataToSecondFragment(CommunicationListener communicationListener) {
        if (communicationListener != null) {
            communicationListener.onDataPassed(name);
        }
    }

    public void setDataToSecondFragment(SecondFragment secondFragment) {
        if (secondFragment != null) {
            secondFragment.onDataPassed(name);
        }
    }
}

File: unit-4/sprint-4/Evaluation/Evaluation/app/src/main/java/com/example/evaluation/views/MainActivity.kt
package com.example.evaluation.views

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.evaluation.R
import com.example.evaluation.Repository.PizzaRepository
import com.example.evaluation.adapter.PizzaAdapter
import com.example.evaluation.api.PizzaApiService
import com.example.evaluation.api.RetrofitHelper
import com.example.evaluation.models.Crust
import com.example.evaluation.viewmodels.CartViewModel
import com.example.evaluation.viewmodels.PizzaViewModel
import com.example.evaluation.viewmodels.PizzaViewModelFactory
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    lateinit var pizzaViewModel: PizzaViewModel
    lateinit var pizzaAdapter: PizzaAdapter
    private val pizzaResponse = mutableListOf<Crust>()
    private lateinit var viewModel: CartViewModel
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        pizzaAdapter = PizzaAdapter(pizzaResponse)
        recyclerView.layoutManager = LinearLayoutManager(this)
        recyclerView.adapter = pizzaAdapter

        val pizzaApiService = RetrofitHelper.getInstance().create(PizzaApiService::class.java)
        val repository = PizzaRepository(pizzaApiService)

        pizzaViewModel = ViewModelProvider(this,PizzaViewModelFactory(repository))
            .get(PizzaViewModel::class.java)
        pizzaViewModel.pizzas.observe(this, Observer {
            pizzaAdapter.
        })

    }
}